systemtests/src/main/java/io/enmasse/systemtest/executor/client/AbstractClient.java

```diff
@@ -0,0 +1,184 @@
+package io.enmasse.systemtest.executor.client;
+
+import io.enmasse.systemtest.Logging;
+import io.enmasse.systemtest.executor.Executor;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.concurrent.*;
+
+/**
+ * Class represent abstract client which keeps common features of client
+ */
+public abstract class AbstractClient {
+    private final int DEFAULT_ASYNC_TIMEOUT = 120000;
+    private final int DEFAULT_SYNC_TIMEOUT = 60000;
+
+    private ClientType clientType;
+    private JsonArray messages;
+    protected ArrayList<String> arguments;
+    protected ArrayList<Argument> allowedArgs;
+
+    /**
+     * Constructor of abstract client
+     * @param clientType type of client
+     */
+    public AbstractClient(ClientType clientType){
+        this.clientType = clientType;
+        this.arguments = new ArrayList<>();

```

Comment:
I prefer initialization during definition, but it's just my opinion. I don't know if it has some performance impact.

Reply:
fixed in 7222b74